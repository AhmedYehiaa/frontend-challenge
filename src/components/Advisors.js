import React from 'react';
import { Table, BackTop, Tooltip } from 'antd';
import styled from '@emotion/styled';

import columns from './TableColumns';

const TableWrapper = styled.div`
  max-width: 75%;
  margin: auto;
`;

const Advisors = ({ advisors, loading, onChange }) => {
  return (
    <TableWrapper>
      <Tooltip placement="topLeft" title="Back to top">
        <BackTop />
      </Tooltip>
      <Table
        columns={columns}
        dataSource={advisors}
        loading={loading}
        onChange={onChange}
        rowKey="id"
        pagination={false}
      />
    </TableWrapper>
  );
}

export default Advisors;