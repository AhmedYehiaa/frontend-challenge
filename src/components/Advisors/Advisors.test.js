import React from "react";
import { shallow, mount } from "enzyme";

import Advisors from "./Advisors";
import { jsxEmptyExpression } from "@babel/types";

describe("Advisors Component", () => {
  let component;
  const advisors = [
    {
      name: "Yehia",
      email: "yehia@email.com",
      avatar: "image.png",
      status: "online",
      language: "english",
      numOfReviews: 50
    }
  ];
  const props = {
    onChange: jest.fn()
  };
  beforeEach(() => {
    component = shallow(
      <Advisors advisors={advisors} loading={false} onChange={props.onChange} />
    );
  });

  it("should render Advisors component", () => {
    expect(component).toMatchSnapshot();
  });

  it("should call onChange function", () => {
    let mountedComponent = mount(
      <advisors advisors={advisors} loading={false} onChange={props.onChange} />
    );
    // mountedComponent.find("input").simulate("change");
    // expect(props.onChange).toHaveBeenCalled();
  });
});
