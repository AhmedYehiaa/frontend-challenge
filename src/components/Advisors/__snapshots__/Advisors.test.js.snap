// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Advisors Component should render Advisors component 1`] = `
<Styled(div)>
  <Tooltip
    arrowPointAtCenter={false}
    autoAdjustOverflow={true}
    mouseEnterDelay={0.1}
    mouseLeaveDelay={0.1}
    placement="topLeft"
    title="Back to top"
    transitionName="zoom-big-fast"
  >
    <BackTop
      visibilityHeight={400}
    />
  </Tooltip>
  <Table
    bordered={false}
    childrenColumnName="children"
    className=""
    columns={
      Array [
        Object {
          "dataIndex": "name",
          "key": "name",
          "title": "Name",
        },
        Object {
          "dataIndex": "avatar",
          "key": "avatar",
          "render": [Function],
          "title": "Avatar",
        },
        Object {
          "dataIndex": "email",
          "key": "email",
          "title": "Email",
        },
        Object {
          "dataIndex": "language",
          "filterMultiple": false,
          "filters": Array [
            Object {
              "text": "English",
              "value": "English",
            },
            Object {
              "text": "Arabic",
              "value": "Arabic",
            },
            Object {
              "text": "German",
              "value": "German",
            },
            Object {
              "text": "Spanish",
              "value": "Spanish",
            },
            Object {
              "text": "Chinese",
              "value": "Chinese",
            },
          ],
          "key": "language",
          "title": "Language",
        },
        Object {
          "dataIndex": "status",
          "filterMultiple": false,
          "filters": Array [
            Object {
              "text": "Online",
              "value": "Online",
            },
            Object {
              "text": "Offline",
              "value": "Offline",
            },
          ],
          "key": "status",
          "render": [Function],
          "title": "Status",
        },
        Object {
          "dataIndex": "numOfReviews",
          "defaultSortOrder": "desc",
          "key": "numOfReviews",
          "sorter": [Function],
          "title": "Reviews",
        },
      ]
    }
    dataSource={
      Array [
        Object {
          "avatar": "image.png",
          "email": "yehia@email.com",
          "language": "english",
          "name": "Yehia",
          "numOfReviews": 50,
          "status": "online",
        },
      ]
    }
    indentSize={20}
    loading={false}
    locale={Object {}}
    onChange={[MockFunction]}
    pagination={false}
    rowKey="id"
    showHeader={true}
    size="default"
    sortDirections={
      Array [
        "ascend",
        "descend",
      ]
    }
    useFixedHeader={false}
  />
</Styled(div)>
`;
